### DISCUSSION TITLE
is it possible to pass the custom, scope or local parameters to winget install when using Microsoft.Winget.DSC/WingetPackage resource

### DISCUSSION THREAD

**Discussion by tristanbarcelon** [microsoft/winget-cli#4985] [Q&A] [answered]
When using winget cli to install packages, it is possible to use the custom, scope or location parameters to provide custom installation parameters or instruct winget cli to pick a properly scoped installer or confine the application installation to a specific folder. The latter is especially useful for applications that consume a lot of disk space over time, such as Visual Studio or SQL Server. [SQLServerDSC exposes multiple parameters for install folders](https://github.com/dsccommunity/SqlServerDsc/blob/85ea1eda6a2d3c3bacd096983d55c99b75dba914/source/Examples/Resources/SqlSetup/8-InstallDefaultInstanceSingleServer2022OrLater.ps1#L69-L77) while VisualStudio does not.

If I wanted to install machine scoped versions of PowerToys or VisualStudioCode, how can I achieve them using Microsoft.Winget.DSC/WingetPackage resource in order to specify machine scope?

**Comment by philippdolder** [comment]
what worked for me with `VSCodium` is to use `winget settings` and then configure preferred scope to `machine`. See https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#scope

Unfortunately I did not yet figure out how to pass custom arguments to make it add the context menu entries

**Comment by denelon** [comment, marked as answer]
Not yet, we've captured the work to support the additional CLI arguments with:
* https://github.com/microsoft/winget-cli/issues/3401
